src/asset/asset.js

```diff
@@ -464,7 +464,7 @@ class Asset extends EventHandler {
     ready(callback, scope) {
         scope = scope || this;
 
-        if (this.resource) {
+        if (this.loaded) {
```

Comment:
Please, ensure logic described above for cubemaps are still present to prevent unnecessary downloads.

Reply:
I've double tested the following:

- Not preloaded, no prefiltered cubemap, on a material environment map that is used in the scene - Only faces are loaded
- Preloaded, no prefiltered cubemap, on a material environment map that is used in the scene - Only faces are loaded
- Not preloaded, prefiltered cubemap generated, on a material environment map that is used in the scene - Only prefiltered data is loaded
- Preloaded, prefiltered cubemap generated, on a material environment map that is used in the scene - Only prefiltered data is loaded

- Not preloaded, no prefiltered cubemap, asset.load called by code - Only faces are loaded
- Preloaded, no prefiltered cubemap, asset.load called by code - Only faces are loaded
- Not preloaded, prefiltered cubemap generated, asset.load called by code - Only prefiltered data is loaded
- Preloaded, prefiltered cubemap generated, asset.load called by code - Only prefiltered data is loaded

- Not preloaded, no prefiltered cubemap, used as skybox, mip level 2 - Only faces are loaded
- Preloaded, no prefiltered cubemap, used as skybox, mip level 2 - Only faces are loaded
- Not preloaded, prefiltered cubemap generated, used as skybox, mip level 2 - Only prefiltered data is loaded
- Preloaded, prefiltered cubemap generated, used as skybox, mip level 2 - Only prefiltered data is loaded
- Not preloaded, prefiltered cubemap generated, used as skybox, mip level 1 - Only faces and prefiltered data are loaded
- Preloaded, prefiltered cubemap generated, used as skybox, mip level 1 - Only faces and prefiltered data are loaded

Have I missed any other possible test cases?

Reply:
I can't see any where in the code base where .ready() is used for cubemap texture loading internally in the engine so we should have the same behaviour as before